### ISSUE TITLE
Performance/ArraySemiInfiniteRangeSlice triggered variable is not an array (but within an array

### ISSUE THREAD

**Issue by karimtimer** [rubocop-hq/rubocop#9121]
I have an array of words. For each word, I remove the first two characters if it matches a conditional. This triggers ArraySemiInfiniteRangeSlice, saying I should use #drop on the element in the loop. This method is not available to Strings and so cannot make that change.

--------

## Expected behavior

Not to raise this cop when dealing with strings. 

## Actual behavior
running rubocop gives this offense:

app/path/to/class.rb:31:15: C: Performance/ArraySemiInfiniteRangeSlice: Use drop instead of [] with semi-infinite range.
`field = field[2..] unless field.match?("total") || field.match?("lob")`

## Steps to reproduce the problem

1.  Prepare an array of string elements. 
2. Loop over the elements. I have used an each loop.
3. Use slice on each element. I have done element[2..]
4. run rubocop.

## RuboCop version

```
$ [bundle exec] rubocop -V
1.4.2 (using Parser 2.7.2.0, rubocop-ast 1.2.0, running on ruby 2.7.2 x86_64-darwin19)
```
